src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -24,63 +25,73 @@ internal class PathHelper
         /// <returns>Normalized path</returns>
         internal static string Normalize(string path)
         {
-            // Get the full path
-            StringBuffer fullPath = new StringBuffer(path.Length);
+            if (path.Length < PathInternal.MaxShortPath)
+            {
+                Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
+                return Normalize(path, new ValueStringBuilder(initialBuffer));
+            }
+            else
+            {
+                return Normalize(path, new ValueStringBuilder());
+            }
+        }
 
+        private static string Normalize(string path, ValueStringBuilder builder)
+        {
+            // Get the full path
             try
             {
-                GetFullPathName(path, ref fullPath);
+                GetFullPathName(path, ref builder);
 
-                if (fullPath.Contains('~'))
+                if (builder.AsSpan().Contains('~'))
                 {
-                    return TryExpandShortFileName(ref fullPath, originalPath: path);
+                    return TryExpandShortFileName(ref builder, originalPath: path);
                 }
                 else
                 {
-                    if (fullPath.Length == path.Length && fullPath.StartsWith(path))
+                    if (builder.AsSpan().Equals(path.AsReadOnlySpan()))
                     {
                         // If we have the exact same string we were passed in, don't bother to allocate another string from the StringBuffer.
                         return path;
                     }
-                    return fullPath.ToString();
+                    return builder.ToString();
                 }
             }
             finally
```

Comment:
I do not think we need to worry about cleanup on exceptional paths.

Reply:
So let the buffer go free and remove the finally? I'm assuming you're referring to better perf without the try block?

Reply:
Right.